client/commands.c

```diff
@@ -48,73 +50,221 @@ void processXferCommand(char *nextParam, FILE *out, int ispwd, uint32_t* pwd);
+    { "state"      , xdag_com_state      , "print the program state" },
+    { "stats"      , xdag_com_stats      , "print statistics for loaded and all known blocks" },
+    { "terminate"  , xdag_com_terminate  , "terminate both daemon and this program" },
+    { "xfer"       ,(xdag_com_func_t)NULL, "[A] xfer S A, transfer S our %s to the address A" },
+    { "help"       , xdag_com_help       , "print xdag commands help" },
+    { (char *)NULL ,(xdag_com_func_t)NULL, (char *)NULL }
+};
+
+int xdag_com_account(char* args, FILE* out) {
+    processAccountCommand(args, out);
+    return 0;
+}
+
+int xdag_com_balance(char * args, FILE* out) {
+    processBalanceCommand(args, out);
+    return 0;
+}
+
+int xdag_com_block(char * args, FILE* out) {
+    processBlockCommand(args, out);
+    return 0;
+}
+
+int xdag_com_lastblocks(char * args, FILE* out) {
+    processLastBlocksCommand(args, out);
+    return 0;
+}
+
+int xdag_com_keyGen(char * args, FILE* out) {
+    processKeyGenCommand(out);
+    return 0;
+}
+
+int xdag_com_level(char * args, FILE* out) {
+    processLevelCommand(args, out);
+    return 0;
+}
+
+int xdag_com_mining(char * args, FILE* out) {
+    processMiningCommand(args, out);
+    return 0;
+}
+
+int xdag_com_net(char * args, FILE* out) {
+    processNetCommand(args, out);
+    return 0;
+}
+
+int xdag_com_pool(char * args, FILE* out) {
+    processPoolCommand(args, out);
+    return 0;
+}
+
+int xdag_com_miners(char * args, FILE* out) {
+    processMinersCommand(args, out);
+    return 0;
+}
+
+int xdag_com_stats(char * args, FILE* out) {
+    processStatsCommand(out);
+    return 0;
+}
+
+int xdag_com_state(char * args, FILE* out) {
+    fprintf(out, "%s\n", get_state());
+    return 0;
+}
+
+int xdag_com_run(char * args, FILE* out) {
+    g_xdag_run = 1;
+    return 0;
+}
+
+int xdag_com_terminate(char * args, FILE* out) {
+    processExitCommand();
+    return -1;
+}
+
+int xdag_com_exit(char * args, FILE* out) {
+    processExitCommand();
+    return -1;
+}
+
+int xdag_com_help(char *args, FILE* out) {
+    printHelp(out);
+    return 0;
+}
+
+char ** xdag_com_completion(const char *text, int start, int end) {
+    char **matches;
+    matches = (char **)NULL;
+    if (start == 0)
+        matches = rl_completion_matches(text, xdag_com_generator);
+    return (matches);
+}
+
+char* dupstr(char* s){
+    char *r;
+    r = malloc(strlen(s) + 1);
+    strcpy(r, s);
+    return (r);
+}
+
+char* xdag_com_generator(const char* text, int state) {
+    static int list_index, len;
+    char *name;
+    if (!state) {
+        list_index = 0;
+        len = strlen(text);
+    }
+    
+    while ((name = commands[list_index].name)) {
+        list_index++;
+        if (strncmp(name, text, len) == 0)
+            return (dupstr(name));
+    }
+    return ((char *)NULL);
+}
+
+XDAG_COMMAND* find_xdag_command(char *name) {
+    register int i;
```

Comment:
Why do you need 'register' modifier here?

Reply:
sorry, copy some code, i think register can delete